src/devices/Vl53L0X/Vl53L0X.cs

```diff
@@ -0,0 +1,1163 @@
+            return sequence;
+        }
+
+        /// <summary>
+        /// Convert sequence step timeout from MCLKs to microseconds with given VCSEL period in PCLKs 
+        /// </summary>
+        /// <param name="timeoutPeriodMclks"></param>
+        /// <param name="vcselPeriodPclks"></param>
+        /// <returns></returns>
+        private UInt32 TimeoutMclksToMicroseconds(UInt32 timeoutPeriodMclks, byte vcselPeriodPclks)
+        {
+            var macroPeriodNanoseconds = CalcMacroPeriod(vcselPeriodPclks);
+            return (UInt32)(((timeoutPeriodMclks * macroPeriodNanoseconds) + (macroPeriodNanoseconds / 2)) / 1000);
+        }
+
+        /// <summary>
+        /// Convert sequence step timeout from microseconds to MCLKs with given VCSEL period in PCLKs 
+        /// </summary>
+        /// <param name="timeoutPeriodMicroseconds">The timeout period in microseconds</param>
+        /// <param name="vcselPeriodPclks">The VCSEL period in PCLKs</param>
+        /// <returns>Returns the converted timeout</returns>
+        private UInt32 TimeoutMicrosecondsToMclks(UInt32 timeoutPeriodMicroseconds, byte vcselPeriodPclks)
+        {
+            var macroPeriodNanoseconds = CalcMacroPeriod(vcselPeriodPclks);
+            return (UInt32)(((timeoutPeriodMicroseconds * 1000) + (macroPeriodNanoseconds / 2)) / macroPeriodNanoseconds);
+        }
+
+        /// <summary>
+        /// Get the VCSEL pulse period in PCLKs for the given period type. 
+        /// </summary>
+        /// <param name="type">The VCSEL period to decode</param>
+        /// <returns>The decoded period</returns>
+        private byte GetVcselPulsePeriod(VcselType type)
+        {
+            switch (type)
+            {
+                case VcselType.VcselPeriodPreRange:
+                    return DecodeVcselPeriod(ReadByte((byte)Registers.PRE_RANGE_CONFIG_VCSEL_PERIOD));
+                case VcselType.VcselPeriodFinalRange:
+                    return DecodeVcselPeriod(ReadByte((byte)Registers.FINAL_RANGE_CONFIG_VCSEL_PERIOD));
+                default:
+                    // Should not arrive
+                    return byte.MaxValue;
+            }
+
+        }
+
+        /// <summary>
+        /// Encode VCSEL pulse period register value from period in PCLKs 
+        /// </summary>
+        /// <param name="periodPclks">The priod in PCLKs</param>
+        /// <returns>the period encoded</returns>
+        private byte EncoreVcselPeriod(byte periodPclks) => (byte)((periodPclks >> 1) - 1);
+
+        /// <summary>
+        /// Decode sequence step timeout in MCLKs from register value 
+        /// format: (LSByte * 2^MSByte) + 1
+        /// </summary>
+        /// <param name="valueToDecode"></param>
+        /// <returns>The decoded value</returns>
+        private UInt32 DecodeTimeout(int valueToDecode) => (UInt32)(((valueToDecode & 0x00FF) << ((valueToDecode & 0xFF00) >> 8)) + 1);
```

Comment:
should this return TimeSpan instead?

Reply:
No, what is used is the raw value after. So better keep it raw. TimeSpan will add complexity and it's not necessary in this case.